src/scripts/main.js

```diff
@@ -1,3 +1,18 @@
+/* eslint-disable no-undef */
+/* eslint-disable padding-line-between-statements */
 'use strict';
+const populationElements = document.querySelectorAll('span.population');
 
-// write your code here
+const populations = Array.from(populationElements)
+  .map((el) => parseInt(el.textContent.replace(/,/g, ''), 10))
+  .filter((num = !isNaN(num)));
```

Comment:
The filter function is incorrectly implemented. The current expression `num = !isNaN(num)` assigns a boolean value to `num` instead of checking if `num` is not NaN. It should be `num => !isNaN(num)` to properly filter out NaN values.